### PR TITLE
Tracing: Add Tempo data source

### PR DESCRIPTION
**Pull request by aocenas**
Adds a Tempo data source mostly by copying Jeager data source code as they should be API compatible.
TODO:
- [x] change image
- [x] change description

Screenshot from 2020-10-13 11-39-54
Screenshot from 2020-10-13 11-39-40

docs image PR https://github.com/grafana/website/pull/2524

### CODE REVIEW

**Review by joe-elliott — APPROVED** [review]

File: public/app/plugins/datasource/tempo/ConfigEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
+import { DataSourceHttpSettings } from '@grafana/ui';
+
+export type Props = DataSourcePluginOptionsEditorProps;
+
+export const ConfigEditor: React.FC<Props> = ({ options, onOptionsChange }) => {
+  return (
+    <DataSourceHttpSettings
+      defaultUrl={'http://localhost:16686'}
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/tempo/ConfigEditor.tsx:10]:
```suggestion
      defaultUrl="http://localhost:16686"
```
nit

File: public/app/plugins/datasource/tempo/QueryField.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export class TempoQueryField extends React.PureComponent<Props> {
+  render() {
+    const { query, onChange } = this.props;
+
+    return (
+      <LegacyForms.FormField
+        label="Trace ID"
+        labelWidth={4}
+        inputEl={
+          <div className={'slate-query-field__wrapper'}>
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/tempo/QueryField.tsx:17]:
```suggestion
          <div className="slate-query-field__wrapper">
```

File: public/app/plugins/datasource/tempo/datasource.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+              preferredVisualisationType: 'trace',
+            },
+          }),
+        ],
+      });
+    }
+  }
+
+  async testDatasource(): Promise<any> {
+    return true;
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/tempo/datasource.ts:79]: shouldn't this try to do a real request?

**Review by zoltanbedi — COMMENTED** [review]
Maybe add it to the menu as well https://github.com/grafana/grafana/blob/master/docs/sources/menu.yaml#L140

File: docs/sources/datasources/tempo.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+aliases = ["/docs/grafana/latest/features/datasources/tempo"]
+[menu.docs]
+name = "Tempo"
+parent = "datasources"
+weight = 800
++++
+
+# Tempo data source
+
+Grafana ships with built-in support for Tempo a high volume, minimal dependency trace storage, OSS tracing solution from Grafana Labs. Just add it as a data source, and you are ready to query your traces in [Explore]({{< relref "../explore/index.md" >}}).
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/tempo.md:15]: Grafana has built-in .... 

Remove "Just" and being the sentence with "Add"

File: docs/sources/datasources/tempo.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+## Adding the data source
+To access Tempo settings, click the **Configuration** (gear) icon, then click **Data Sources**, and then click **Tempo**.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/tempo.md:18]: To access Tempo settings, click **Configuration** (gear) icon, then click **Data Sources** > **Tempo**.

File: docs/sources/datasources/tempo.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+| Name            | Description                                                                                                                                   |
+| --------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| _Name_          | The data source name. This is how you refer to the data source in panels, queries, and Explore.                                                 |
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/tempo.md:22]: The data source name using which you will refer to ....

**Review by achatterjee-grafana — COMMENTED** [review]
Added a few copy-edit suggestions.

File: public/app/plugins/datasource/tempo/datasource.ts (hunk 3 above)
Comment (aocenas) [reply public/app/plugins/datasource/tempo/datasource.ts:88]: yeah forgot about it in jaeger, added a simple test there

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Pull request merged**